trieste/models/gpflow/sampler.py

```diff
@@ -101,7 +105,7 @@ class BatchReparametrizationSampler(ReparametrizationSampler):
     form a continuous curve.
     """
 
-    def __init__(self, sample_size: int, model: ProbabilisticModel):
+    def __init__(self, sample_size: int, model: SupportsPredictJoint):
```

Comment:
perhaps we should add a runtime check that the model is instance of SupportsPredictJoint, same as in acquisition functions?